frontend/app/components/wp-buttons/wp-list-view-button/wp-list-view-button.directive.ts

```diff
@@ -26,55 +26,51 @@
 // See doc/COPYRIGHT.rdoc for more details.
 // ++
 
-angular
-  .module('openproject.wpButtons')
-  .directive('wpListViewButton', wpListViewButton);
+import {WorkPackageNavigationButtonController, wpButtonDirective} from '../wp-buttons.module';
 
-function wpListViewButton() {
-  return {
-    restrict: 'E',
-    templateUrl: '/components/wp-buttons/wp-list-view-button/wp-list-view-button.directive.html',
+export class WorkPackageListViewButtonController extends WorkPackageNavigationButtonController {
+  public projectIdentifier:number;
 
-    scope: {
-      projectIdentifier: '='
-    },
+  public accessKey:number = 7;
+  public activeState:string = 'work-packages.list';
+  public labelKey:string = 'js.button_list_view';
+  public buttonId:string = 'work-packages-list-view-button';
+  public iconClass:string = 'icon-view-list';
 
-    bindToController: true,
-    controller: WorkPackageListViewButtonController,
-    controllerAs: 'vm'
-  };
-}
+  constructor(public $state:ng.ui.IStateService, public I18n) {
+    'ngInject';
 
-function WorkPackageListViewButtonController($state, inplaceEditAll) {
-  var vm = this;
+    super($state, I18n);
+  }
 
-  angular.extend(vm, {
-    isActive: function () {
-      return $state.is('work-packages.list');
-    },
+  public isActive() {
+    return this.$state.is(this.activeState);
+  }
 
-    openListView: function () {
-      var params = {
-        projectPath: vm.projectIdentifier
-      };
+  public performAction() {
+    this.openListView();
+  }
 
-      angular.extend(params, $state.params);
-      $state.go('work-packages.list', params);
-    },
+  public openListView() {
+    var params = {
+      projectPath: this.projectIdentifier
+    };
 
-    isDisabled: function () {
-      return inplaceEditAll.state;
-    },
+    angular.extend(params, this.$state.params);
+    this.$state.go(this.activeState, params);
+  }
+}
 
-    getAccessKey: function () {
-      if (!vm.isActive()) return 8;
```

Comment:
Yes, I figured that because 8 is already used by another component, 7 would be the better choice.
Although I might be wrong. I don't know what was specified - just added what seemed to be sensible.